### DISCUSSION TITLE
Navigation Principles

### DISCUSSION THREAD

**Discussion by scruffian** [WordPress/gutenberg#45032] [Developer Experience]
The navigation block and the parts of the project that surround it have become increasingly difficult to work with. Here are some principles that might help guide the decisions we make in this project going forward:

1. When we make changes to the block we should be reasonably confident that we didn’t introduce unexpected bugs (through useful tests).
2. The complexity of the block should be encapsulated in ways that make it easy to understand for someone who is new to it.
3. We have a clear separation of concerns between different parts of the block, so that changes in one part don’t have unexpected consequences in other parts.
4. Features of this block should be created in such a way that they can be used by other blocks if they want them.
5. Fixing bugs is critical to the development of new features. Unfixed bugs lead to further complexity when we try to implement new features. Therefore fixing bugs should always be a precondition for new development.

These ideas were developed in collaboration with @getdave @MaggieCabrera @draganescu and @ajlende

**Comment by getdave** [comment]
Now linked up with the [Tracking Issue](https://github.com/WordPress/gutenberg/issues/38275)
